src/modules/stats/overwatch-stats.ts

```diff
@@ -0,0 +1,189 @@
+			memberName: 'overwatch-stats',
+			name: 'overwatch-stats',
+			throttling: {
+				duration: 3,
+				usages: 2
+			}
+		});
+	}
+
+	/**
+	 * Run the "overwatch-stats" command.
+	 *
+	 * @param {CommandoMessage} msg
+	 * @param {OverwatchStatsArguments} args
+	 * @returns {(Promise<Message | Message[]>)}
+	 * @memberof BrewCommand
+	 */
+	public async run(msg: CommandoMessage, args: OverwatchStatsArguments): Promise<Message | Message[]> {
+		startTyping(msg);
+
+		// Replace any hashtag with a dash for proper api formatting.
+		args.battletag = args.battletag.replace('#', '-');
+
+		// Make the api call.
+		try {
+			await getProfile(args.platform, args.region, args.battletag, (error: Error, profile: Profile) => {
+				stopTyping(msg);
+
+				if (error) {
+					msg.say(`The api returned the following error: ${error.message}.`);
+				} else if (profile) {
+					const overwatchEmbed: MessageEmbed = new MessageEmbed({
+						author: {
+							icon_url: profile.competitive ? profile.competitive.rank_img : null,
+							name: args.battletag.replace('-', '#'),
+							url: `https://playoverwatch.com/career/${args.platform}/${args.battletag}`
+						},
+						fields: [
+							{
+								inline: true,
+								name: 'Level',
+								value: profile.level.toString()
+							},
+							{
+								inline: true,
+								name: 'Rank',
+								value: profile.competitive.rank ? profile.competitive.rank.toString() : 'N/A'
+							},
+							{
+								inline: true,
+								name: 'Competitive Playtime',
+								value: profile.playtime.competitive || '0'
+							},
+							{
+								inline: true,
+								name: 'Competitive Win Rate',
+								value: profile.games.competitive.win_rate ? `${profile.games.competitive.win_rate}%` : 'N/A'
+							},
+							{
+								inline: true,
+								name: 'Quickplay Playtime',
+								value: profile.playtime.quickplay || '0'
+							},
+							{
+								inline: true,
+								name: 'Quickplay Games Won',
+								value: profile.games.quickplay.won ? profile.games.quickplay.won.toString() : '0'
+							}
+						],
+						thumbnail: {
+							url: profile.portrait
+						}
+					});
+					
+					if (profile.endorsement) {
+						overwatchEmbed.fields.push(
+							{
+								inline: true,
+								name: 'Endorsement Level',
+								value: profile.endorsement.level.toString()
+							},
+							{
+								inline: true,
+								name: 'Sportsmanship',
+								value: `${profile.endorsement.sportsmanship.rate}%`
+							},
+							{
+								inline: true,
+								name: 'Shotcaller',
+								value: `${profile.endorsement.shotcaller.rate}%`
+							},
+							{
+								inline: true,
+								name: 'Good Teammate',
+								value: `${profile.endorsement.teammate.rate}%`
+							}
+						);
+					}
+					deleteCommandMessages(msg, this.client);
+
+					return msg.embed(overwatchEmbed);
+				} else {
+					return msg.say('Something is weird.. There was no profile or errors found.');
+				}
+			})
+
+		} catch(exception) {
```

Comment:
This is not how we do promises; we have a global error handler.

Remove the try/catch block.